torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3529,7 +3530,15 @@ struct to_ir {
       case TK_IS:
       case TK_ISNOT:
       case TK_AND:
-      case TK_OR:
+      case TK_OR: {
+        auto lhs = emitSugaredExpr(Expr(tree->tree(0)), 0)
+                       ->asValue(tree->tree(0)->range(), method);
+        auto rhs = emitSugaredExpr(Expr(tree->tree(1)), 0)
+                       ->asValue(tree->tree(1)->range(), method);
```

Comment:
`lhs` and `rhs` types should be checked to ensure they match (unless in a boolean context)